src/payment/models/bank-tx/olky-pay.service.ts

```diff
@@ -0,0 +1,181 @@
+import { Injectable, ServiceUnavailableException } from '@nestjs/common';
+import { Interval } from '@nestjs/schedule';
+import { Method } from 'axios';
+import { Config } from 'src/config/config';
+import { SettingService } from 'src/shared/models/setting/setting.service';
+import { HttpError, HttpService } from 'src/shared/services/http.service';
+import { Util } from 'src/shared/util';
+import { BankTxBatch } from './bank-tx-batch.entity';
+import { BankTx, BankTxIndicator, BankTxType } from './bank-tx.entity';
+import { BankTxService } from './bank-tx.service';
+
+export interface Transaction {
+  idCtp: number;
+  dateEcriture: number[];
+  dateValeur: number[];
+  codeInterbancaireInterne: TransactionType;
+  codeInterbancaire: string;
+  credit: number;
+  debit: number;
+  line1: string;
+  line2: string;
+  instructingIban: string;
+}
+
+export interface Balance {
+  balance: number;
+  balanceOperationYesterday: number;
+}
+
+enum TransactionType {
+  RECEIVED = 'SCT_RECEIVED',
+  SENT = 'SCT_SENT',
+  BILLING = 'BILLING',
+}
+
+@Injectable()
+export class OlkyPayService {
+  private readonly baseUrl = 'https://ws.olkypay.com/reporting/';
+  private readonly loginUrl = 'https://stp.olkypay.com/auth/realms/b2b/protocol/openid-connect/token';
+  private accessToken = 'access-token-will-be-updated';
+
+  constructor(
+    private readonly http: HttpService,
+    private readonly bankTxService: BankTxService,
+    private readonly settingService: SettingService,
+  ) {}
+
+  // --- TRANSACTION HANDLING --- //
+  @Interval(60000)
+  async checkInputs(): Promise<void> {
+    const settingKey = 'lastOlkyPayDate';
+    const lastModificationTime = await this.settingService.get(settingKey);
+    const newModificationTime = new Date().toISOString().split('T')[0];
+
+    const allTransaction = await this.getTransactions(lastModificationTime);
+
+    for (const transaction of allTransaction) {
+      try {
+        const bankTx = this.parseTransaction(transaction);
+        await this.bankTxService.create(bankTx);
+      } catch (e) {
+        console.error(`Failed to import Transaction:`, e);
+      }
+    }
```

Comment:
It will spam our log with errors, if we don't check if the transaction already exists in DB